File: client/src/app/modules/tiers/model/ITiers.ts
import { AccountingAccount } from "../../accounting/model/AccountingAccount";
import { City } from "../../miscellaneous/model/City";
import { Country } from "../../miscellaneous/model/Country";
import { IAttachment } from "../../miscellaneous/model/IAttachment";
import { IDocument } from "../../miscellaneous/model/IDocument";
import { Language } from "../../miscellaneous/model/Language";
import { Mail } from "../../miscellaneous/model/Mail";
import { Phone } from "../../miscellaneous/model/Phone";
import { TiersFollowup } from "../../miscellaneous/model/TiersFollowup";
import { Employee } from "../../profile/model/Employee";

export interface ITiers extends IAttachment, IDocument {
  id: number;
  salesEmployee: Employee | undefined;
  formalisteEmployee: Employee | undefined;
  insertionEmployee: Employee | undefined;
  mailRecipient: string | null;
  language: Language;
  address: string;
  postalCode: string;
  cedexComplement: string;
  city: City;
  country: Country;
  mails: Mail[];
  phones: Phone[];
  tiersFollowups: TiersFollowup[];
  observations: string;
  accountingAccountCustomer: AccountingAccount;
  accountingAccountProvider: AccountingAccount;
  accountingAccountDeposit: AccountingAccount;
}

File: src/main/java/com/jss/osiris/modules/tiers/model/ITiers.java
package com.jss.osiris.modules.tiers.model;

import java.util.List;

import com.jss.osiris.modules.accounting.model.AccountingAccount;
import com.jss.osiris.modules.miscellaneous.model.City;
import com.jss.osiris.modules.miscellaneous.model.Country;
import com.jss.osiris.modules.miscellaneous.model.IDocument;
import com.jss.osiris.modules.miscellaneous.model.IId;
import com.jss.osiris.modules.miscellaneous.model.Language;
import com.jss.osiris.modules.miscellaneous.model.Mail;
import com.jss.osiris.modules.miscellaneous.model.Phone;
import com.jss.osiris.modules.profile.model.Employee;

public interface ITiers extends IId, IDocument {

	public Employee getSalesEmployee();

	public Employee getFormalisteEmployee();

	public Employee getInsertionEmployee();

	public String getMailRecipient();

	public Language getLanguage();

	public String getAddress();

	public String getPostalCode();

	public City getCity();

	public Country getCountry();

	public String getObservations();

	public List<Mail> getMails();

	public List<Phone> getPhones();

	public Boolean getIsIndividual();

	public AccountingAccount getAccountingAccountProvider();

	public AccountingAccount getAccountingAccountCustomer();

	public AccountingAccount getAccountingAccountDeposit();

}
